nx/lib/nx.ex

```diff
@@ -14312,7 +14312,27 @@ defmodule Nx do
 
     shape = Nx.Shape.take_along_axis(tensor.shape, indices.shape, axis)
 
-    result = impl!(tensor).take_along_axis(%{tensor | shape: shape}, tensor, indices, axis)
+    out = %{tensor | shape: shape}
+
+    result =
+      Nx.Shared.optional(:take_along_axis, [tensor, indices, axis], out, fn tensor,
+                                                                            indices,
+                                                                            axis ->
```

Comment:
All arguments to optional must tensors or options, so you can fix it like this:

```suggestion
      Nx.Shared.optional(:take_along_axis, [tensor, indices, [axis: axis]], out, fn
        tensor, indices, _opts ->
```